flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorPartitionLifecycleTest.java

```diff
@@ -194,7 +183,8 @@ private void testJobMasterConnectionTerminationAfterExternalReleaseOrPromotion(T
 				TaskExecutorPartitionInfo.from(resultPartitionDeploymentDescriptor)));
 
 			final CompletableFuture<Collection<ResultPartitionID>> firstReleasePartitionsCallFuture = new CompletableFuture<>();
-			runInTaskExecutorThreadAndWait(taskExecutor, () -> shuffleEnvironment.releasePartitionsLocallyFuture = firstReleasePartitionsCallFuture);
+			runInTaskExecutorThreadAndWait(taskExecutor, () -> partitionTracker.setStopTrackingAndReleasePartitionsConsumer(firstReleasePartitionsCallFuture::complete));
+			runInTaskExecutorThreadAndWait(taskExecutor, () -> partitionTracker.setIsTrackingPartitionsForFunction(jobId -> true));
```

Comment:
I think it would be a bit nicer to move these test arrangements to the top of the test. One could even set them when creating the `TestingTaskExecutorPartitionTracker` (e.g. via a builder). Then one would not have to run these things in the task executor thread and one would not mix act logic with arrange logic.